### PR TITLE
Updating Microsoft.Security swagger json file - Added two missing properties

### PR DESCRIPTION
**Pull request by orendd**
…ng properties. These properties existed in the API but were forgotten when the swagger json was created

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json [hunk 1]
```diff
@@ -2500,16 +2500,27 @@
...
+                "allowedSourceAddressPrefixes": {
+                    "type": "array",
+                    "description": "Mutually exclusive with the \"allowedSourceAddressPrefix\" parameter. If both are omitted, the request will be for the source IP address of the initiate request.",
+                    "items": {
+                        "type": "string",
+                        "description": "IP address or CIDR, for example \"192.168.0.3\" or \"192.168.0.0/16\"."
+                    }
+                },
+                "duration": {
+                    "type": "string",
```
Comment (chlahav) [inline specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json]:
```suggestion
                    "type": "string",
                    "format": "iso8601",
```

**Review by chlahav — APPROVED** [review]

**Comment by praries880** [comment]
@orendd @chlahav 
I wanna understand the scenario where you want to support both "allowedSourceAddressPrefix" and "allowedSourceAddressPrefixes"
Are you supporting both the singular and the multiple (list) versions of the property just so that you dont break your customers or is there a valid user case scenario to support both of these at the same time?

File: specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json (hunk 1 above)
Comment (orendd) [reply specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json]: Seems like this format is not support by some of the tools and caused the validation to fail.

**Comment by orendd** [comment]
@praries880 
This API is already applied in Prod since many months ago, and these changes only add the missing properties that were mistakenly omitted when the json file was originally written.

To answer your question, yes. The `allowedSourceAddressPrefixes` property was added after `allowedSourceAddressPrefix` to support an array of values, and we chose to keep the old property to avoid any breaking changes for our customers.
The description for these values mention that they are mutually exclusive, meaning that populating both of them will fail the request.

File: specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json [hunk 2]
```diff
@@ -2500,16 +2500,28 @@
...
+                    "format": "iso8601",
```
Comment (orendd) [inline specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json]:
```suggestion
```

**Comment by praries880** [comment]
@orendd 
Thanks for replying.

WRT the breaking change point… Why not just bump up the API version and take the singular field out in the new API version. On the server side you can still support both the new and the old API version. As it currently stands it’s a very confusing API to have both the singular and array version of the field in the same API. Given this is still a preview version of the API I think breaking changes are fine.

@johanste  I would like your input on this as well.. what do you think is the best way forward?

**Comment by johanste** [comment]
A new API version is (should) be the most straight-forward way of ensuring that customers are not broken while evolving the API. 

If we don't want customers to use the old attributes, then simply removing them from the swagger would also be an option (the service would continue to support them in requests, and it would continue to return them in responses). While this would imply a new major version of the SDK (if the SDK was released as non-preview, and it was following semver), it would not affect existing clients until they updated their SDK reference. And unless we want to continue supporting both options indefinitely, we'd have to remove them *at some point*. 

So, my suggestion is to consider removing the old attributes in the swagger now.

**Comment by orendd** [comment]
@praries880, @johanste, thanks for your input 😊
I agree with your points, seems like the ideal solution. However, on the technical side I think it will take some effort to support both versions on the server side, which we will need to prioritize against other tasks. For example, assuming we can configure two api versions to the same endpoint, some error codes may seem confusing as they would indicate a missing parameter that is not even supported by the API. I can think about a fairly clean solution in the code to enable this, but as mentioned earlier it will require some prioritization which could delay this solution.

To compromise for that, I’d prefer to fix the current swagger file to match the existing API, while promoting the new API in our documentation (show the preferred properties to use).
In the meantime, we can work on implementing a new API on the server side 😊

Let me know your thoughts on my suggestion!

Thanks,
Oren

**Comment by orendd** [comment]
@praries880, @johanste, friendly ping on above

**Comment by johanste** [comment]
I can't really estimate the cost of supporting two API versions on the service vs. supporting duplication of the properties of the attribute :). I would have suspected that you would have to spend roughly the same amount of time ensuring that clients using the "old" property gets the appropriate error message/don't immediately break irrespective of if they are in the same API version or in two different API versions, but again, that is my uninformed assumption. 

In addition, if it was my service, I would probably have preferred to do the breaking SDK change within the preview API version instead of breaking between the preview and stable versions. 

This is all the gentle push-back I'm going to given. @praries880, do you have any additional concerns?

**Comment by praries880** [comment]
I concur with @johanste 's feedback/assesment above.

**Comment by praries880** [comment]
@orendd Any updates on this?

**Comment by orendd** [comment]
@johanste , @praries880 
I apologize for not replying for a long time since your last comments.
Nothing has really changed on our part in the meantime.
As mentioned before, I agree with all your points but at the moment there's no one to support these changes. I prefer to at least improve the documentation/API (while still supporting the existing functionality) as a best effort.
If your opinions still stand, please let me know so we can set up a short meeting to clarify my point.
Thanks!

**Comment by ravbhatnagar** [comment]
ARM Feedback - a new API version which just supports one property would be the most ideal option but since service team wants to bring parity with what is already present/deployed on the service, its fine although its not ideal API design. Signing off from ARM side,

**Comment by salameer** [comment]
@praries880 would u kindly follow up on this? it seems that this should go in based on the last comments

**Comment by salameer** [comment]
ping @praries880

**Comment by praries880** [comment]
@orendd can you kindly resolve the conflicts and ill merge this PR

**Comment by salameer** [comment]
Closing since this is stable for a while @orendd please request re-opening if still required

**Pull request closed without merging**